packages/react-instantsearch/src/components/classNames.js

```diff
@@ -2,11 +2,19 @@ import cx from 'classnames';
 const prefix = 'ais';
 
 export default function classNames(block) {
-  return (...elements) => ({
-    className: cx(
-      elements
-        .filter(element => element !== undefined && element !== false)
-        .map(element => `${prefix}-${block}__${element}`)
-    ),
+  return (bemElements, otherElements) => ({
```

Comment:
I am not a huge fan of the fact that `classNames` function return an object. IMO it will be much easier to return a string and apply this string to the className property. With this kind of return type we can't use it with outside of the `jsx` scope. We can't easily do something like:

```js
const myClassName = cx(['myClass', isTrue && 'myOtherClass']); // ais-Block-myClass ais-Block-myOtherClass
```